api/src/migrations/geoprocessing/1611221157285-initialGeoDBSetup.ts

```diff
@@ -0,0 +1,185 @@
+      );
+
+      CREATE TYPE "adm_level" AS ENUM (
+        0,
+        1,
+        2
+      );
+
+-- Administrative regions table.
+      CREATE TABLE "admin_regions" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "ogc_fid" int NOT NULL,
+        "the_geom" geometry CONSTRAINT admin_regions_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "name_0" varchar,
+        "name_1" varchar,
+        "name_2" varchar,
+        "iso3" varchar(3),
+        "gid_0" varchar,
+        "gid_1" varchar,
+        "gid_2" varchar,
+        "level" adm_level NOT NULL,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      ) PARTITION BY LIST(level);
+
+      CREATE TABLE admin_regions_0 PARTITION OF admin_regions FOR VALUES IN (0);
+      CREATE TABLE admin_regions_1 PARTITION OF admin_regions FOR VALUES IN (1);
+      CREATE TABLE admin_regions_2 PARTITION OF admin_regions FOR VALUES IN (2);
+
+      -- Protected Areas table.
+      CREATE TABLE "wdpa" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "wdpaid" float8,
+        "the_geom" geometry CONSTRAINT wdpa_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "full_name" varchar,
+        "iucn_cat" varchar,
+        "shape_leng" float8,
+        "shape_area" float8,
+        "iso3" varchar(3) REFERENCES "admin_regions" ("iso3"),
+        "status" text,
+        "desig" text,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Features data table (will hold species and bioma data).
+      CREATE TABLE "features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "properties" jsonb,
+        "source" source_type
+      );
+
+      -- Usage of features data within scenarios table (will hold species and bioma data).
+      CREATE TABLE "scenario_features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "feature_class_id" uuid,
+        "scenario_id" uuid,
+        "total_area" varchar,
+        "admin_area" timestamp,
+        "current_pa" source_type,
+        "spf" float8,
+        "target" float8,
+        "target2" float8,
+        "targetocc" float8,
+        "sepnum" float8,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Planning units geometry data table.
+      CREATE TABLE "planing_units_geom" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry UNIQUE NOT NULL CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "type" shape_type,
+        "size" int
+      ) PARTITION BY LIST (type);
```

Comment:
this won't work, but here the issue is slightly different from the other partition on `admin_regions`, because of the `UNIQUE` constraint on the `the_geom` column, combined with the fact that we may not be able to use a hash partition by hash of `the_geom` because

```
ERROR:  operator class "hash_geometry_ops" of access method hash is missing support function 2 for type geometry
```

(and I doubt hash partitioning would be useful here in any case)...

I think we should try to find a different way to partition this - eventually. For the moment, using `PRIMARY KEY (id, type)` should work.